plugins/guests/suse/cap/change_host_name.rb

```diff
@@ -5,11 +5,10 @@ class ChangeHostName
         def self.change_host_name(machine, name)
           comm = machine.communicate
 
-          if !comm.test("hostname -f | grep '^#{name}$'", sudo: false)
+          if !comm.test("getent hosts '#{name}'", sudo: false)
```

Comment:
This does not query the hostname though, this only queries the settings from `/etc/hosts`. `hostnamectl status --static` will give you the current full hostname.

Reply:
Thanks for your comments!

`hostname status --static` will print the hostname, not the FQDN which depends on the resolver. And this is what `hostname -f` is doing: It uses the hostname, gives it to the resolver and returns the result. 
`getent hosts '#{name}'` is the command I've found which seems to be the closest to `hostname -f`, as on the base box the resolver will use `/etc/hosts`.

Of course one could discuss if this test is really necessary or if comparing the hostname (basename) would be sufficient. I guess the original authors wanted vagrant to be able to change the domain part of the box' fqdn after the initial start.

Reply:
> Of course one could discuss if this test is really necessary or if comparing the hostname (basename) would be sufficient. I guess the original authors wanted vagrant to be able to change the domain part of the box' fqdn after the initial start.

I think this merely tries to find out if vagrant already changed the hostname of the box, so it makes sense to query the boxes' fqdn instead of the hostname.

Thanks for clarifying this!